plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPreferencesModel.java

```diff
@@ -0,0 +1,101 @@
+package com.google.cloud.tools.eclipse.appengine.deploy.ui;
+
+import org.eclipse.core.resources.IProject;
+import org.osgi.service.prefs.BackingStoreException;
+
+import com.google.cloud.tools.eclipse.appengine.deploy.standard.StandardDeployPreferences;
+
+public class DeployPreferencesModel {
+
+  private StandardDeployPreferences preferences;
+
+  private boolean promptForProjectId;
+  private String projectId;
+  private boolean overrideDefaultVersioning;
+  private String version;
+  private boolean autoPromote;
+  private boolean overrideDefaultBucket;
+  private String bucket;
+
+  public DeployPreferencesModel(IProject project) {
+    preferences = new StandardDeployPreferences(project);
+    loadPreferences();
+  }
+
+  public void savePreferences() throws BackingStoreException {
+    preferences.setProjectId(getProjectId());
+    preferences.setPromptForProjectId(isPromptForProjectId());
+    preferences.setOverrideDefaultVersioning(isOverrideDefaultVersioning());
+    preferences.setVersion(getVersion());
+    preferences.setAutoPromote(isAutoPromote());
+    preferences.setOverrideDefaultBucket(isOverrideDefaultBucket());
+    preferences.setBucket(getBucket());
+    preferences.save();
+  }
+
+  private void loadPreferences() {
+    setProjectId(preferences.getProjectId());
+    setPromptForProjectId(preferences.isPromptForProjectId());
+    setOverrideDefaultVersioning(preferences.isOverrideDefaultVersioning());
+    setVersion(preferences.getVersion());
+    setAutoPromote(preferences.isAutoPromote());
+    setOverrideDefaultBucket(preferences.isOverrideDefaultBucket());
+    setBucket(preferences.getBucket());
+  }
+
+  public boolean isPromptForProjectId() {
+    return promptForProjectId;
+  }
+
+  public void setPromptForProjectId(boolean promptForProjectId) {
+    this.promptForProjectId = promptForProjectId;
+  }
+
+  public String getProjectId() {
+    return projectId;
+  }
+
+  public void setProjectId(String projectId) {
```

Comment:
I'd keep any validations separate from this class. `null` wouldn't cause a problem in the code (as in no NPE being thrown).